pkg/virt-controller/watch/application.go

```diff
@@ -268,6 +271,9 @@ func (vca *VirtControllerApp) AddFlags() {
 	flag.StringVar(&vca.launcherImage, "launcher-image", launcherImage,
 		"Shim container for containerized VMs")
 
+	flag.StringVar(&vca.imagePullSecret, "image-pull-secret", imagePullSecret,
```

Comment:
For a KubeVirt user to take advantage of this option, they would have to manually edit the kubevirt manifest before deploying kubevirt.

If we're going to provide a user defined variable like this, we should probably be putting it in a configmap instead of a cli argument. The idea of a KubeVirt configmap that lets users dynamically set/modify values like "image-pull-secret" is something that we're interested in doing soon.

What would you think about dropping the cli option for now and just making the imagePullSecret feature something that has to be set on the volume for now? Then, once we get the dynamic configmap implemented we introduce the ablity to set a global pull secret?

This assumes the pull secret isn't somehow required for the virt-launcher container itself. If that's the case, then what you've done here makes sense now since we define where to get the virt-launcher container image as a cli arg already.

Reply:
in that case, it makes complete sense what you've done.